### ISSUE TITLE
linter: react-hooks/rules-of-hooks false positive

### ISSUE THREAD

**Issue by DreierF** [oxc-project/oxc#9795]
### What version of Oxlint are you using?

0.15.15

### What command did you run?

pnpm exec oxlint

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "plugins": [
    "typescript",
    "react"
  ],
  "categories": {
    "correctness": "off"
  },
  "rules": {
    "react-hooks/rules-of-hooks": "error"
  }
}

```

### What happened?

```tsx
function Component() {
	if (!useHasPermission()) {
		return null;
	}
	return <Content />;
}
```

Code like this causes a FP: `React Hook "useHasPermission" is called conditionally. React Hooks must be called in the exact same order in every component render.`

The order in which the hook is called is always the same and is also understood by the corresponding ESLint plugin.
